angular2-jwt.ts

```diff
@@ -378,11 +371,21 @@ export class AuthModule {
     }
   }
 
-  static forRoot(config: AuthConfig): ModuleWithProviders {
+  static forRoot(config?: IAuthConfigOptional): ModuleWithProviders {
     return {
       ngModule: AuthModule,
       providers: [
-        {provide: AuthConfig, useValue: config}
+        {provide: AUTH_CONFIGURATION, useValue: config ? config : {}},
```

Comment:
I know, but you can't call a "new" or anything which is like a function call.
I "copied" the logic from https://github.com/angular/angular/blob/42cf06fa12924882d18d1167432214fdc66b1158/modules/%40angular/router/src/router_module.ts#L148 there is a reason why they use this.
As i mentioned above i also recomment using `AuthConfig` as useValue but it wasn't possible.
I tried many things...